src/Workspaces/Core/Portable/Utilities/NullableHelpers/NullableExtensions.cs

```diff
@@ -8,31 +8,9 @@ namespace Microsoft.CodeAnalysis
 {
     internal static partial class NullableExtensions
     {
-        public static T WithNullability<T>(this T typeSymbol, NullableFlowState flowState) where T : class, ITypeSymbol
-        {
-            // TODO: call the compiler API once it's available
-            switch (flowState)
-            {
-                case NullableFlowState.None:
-                    return (T)typeSymbol.WithNullableAnnotation(NullableAnnotation.None);
-                case NullableFlowState.NotNull:
-                    return (T)typeSymbol.WithNullableAnnotation(NullableAnnotation.NotAnnotated);
-                case NullableFlowState.MaybeNull:
-                    return (T)typeSymbol.WithNullableAnnotation(NullableAnnotation.Annotated);
-                default:
-                    throw ExceptionUtilities.UnexpectedValue(typeSymbol);
-            }
-        }
-
-        public static ITypeSymbol? GetConvertedTypeWithFlowNullability(this TypeInfo typeInfo)
-            => typeInfo.ConvertedType?.WithNullability(typeInfo.ConvertedNullability.FlowState);
-
         public static ITypeSymbol? GetConvertedTypeWithAnnotatedNullability(this TypeInfo typeInfo)
             => typeInfo.ConvertedType?.WithNullableAnnotation(typeInfo.ConvertedNullability.Annotation);
```

Comment:
I'm not deleting this and the other helper in this PR, as the remaining use appears to be a tricky case in extract method. It wasn't clear at first glance whether or not it's doing unnecessary work and using the flow nullability is actually correct, or whether this is actually a very special case where we do need to ignore the flow nullability.